obd/obd.py

```diff
@@ -207,13 +213,46 @@ def print_commands(self):
         for c in self.supported_commands:
             print(str(c))
 
-
+    def print_discovered(self):
+        """
+            Utility function meant to print all information discovered:
+            protocol, port name, port baudrate and all supported commands.
+        """
+        if self.interface is not None:
+            print ("The following settings were used to connect to the ECU:")
+            print ("Protocole: " + self.get_protocol_name())
+            print ("Port name: " + self.get_port_name())
+            print ("Port rate: " + self.get_port_baudrate())
+            print ("The following OBD commands are supported:")
+
+            _mylist=[]
+            for c in self.supported_commands:
+                _mylist.append(str(c))
+            _mylist.sort()
+            for i in _mylist:
+                print (i)
+            
+        else:
+            print ("Impossible to print discovered information: no connection to the ECU.")
+            
```

Comment:
Hey, thanks for the PR!

While I don't have a problem with the printing code, I don't think the library core is the right place for this. I'd like to separate the concerns of OBD communications and tool-like terminal printing. I've been meaning to remove the `print_commands` function above (which I've intentionally not documented). Instead, I'd be totally supportive of adding a /tools directory for CLI applications akin to this.